src/main/kotlin/com/sourcegraph/cody/chat/ui/LLMModelDropdown.kt

```diff
@@ -0,0 +1,73 @@
+package com.sourcegraph.cody.chat.ui
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.invokeLater
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.ComboBox
+import com.intellij.ui.MutableCollectionComboBoxModel
+import com.intellij.util.concurrency.annotations.RequiresEdt
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol.ChatModelsParams
+import com.sourcegraph.cody.agent.protocol.ChatModelsResponse
+import com.sourcegraph.cody.chat.SessionId
+import com.sourcegraph.cody.config.CodyAccount.Companion.isEnterpriseAccount
+import com.sourcegraph.cody.config.CodyAuthenticationManager
+import com.sourcegraph.cody.ui.ChatModel
+import com.sourcegraph.cody.ui.LLMModelComboBoxItem
+import com.sourcegraph.cody.ui.LLMModelComboBoxRenderer
+import com.sourcegraph.common.BrowserOpener
+import java.util.concurrent.TimeUnit
+
+class LLMModelDropdown(val project: Project, initiallySelected: ChatModel?) :
+    ComboBox<LLMModelComboBoxItem>(MutableCollectionComboBoxModel()) {
+
+  init {
+    val activeAccountType = CodyAuthenticationManager.instance.getActiveAccount(project)
+    if (activeAccountType.isEnterpriseAccount()) {
+      isEnabled = false
+    }
+
+    renderer = LLMModelComboBoxRenderer()
+    if (initiallySelected != null) {
+      addItem(LLMModelComboBoxItem(initiallySelected.icon, initiallySelected.displayName))
+    }
+  }
+
+  fun fetchAndUpdateModels(sessionId: SessionId) {
+    CodyAgentService.withAgent(project) { agent ->
+      val chatModels = agent.server.chatModels(ChatModelsParams(sessionId))
+      val response = chatModels.completeOnTimeout(null, 4, TimeUnit.SECONDS).get()
```

Comment:
I think `4 sec` might not be enough for the first chat which appears right after the startup (when agent is also starting up).

Reply:
`fetchAndUpdateModels` happens only when we receive the session id.

That ☝️ makes this timeout series (not parallel) to the server start up.